crates/oxc_linter/src/service.rs

```diff
@@ -167,21 +170,39 @@ impl Runtime {
             return;
         };
         let Some(source_type_and_text) = Self::get_source_type_and_text(path, ext) else { return };
-        let (source_type, source_text) = match source_type_and_text {
+        let (source_type, original_source_text) = match source_type_and_text {
             Ok(source_text) => source_text,
             Err(e) => {
                 tx_error.send(Some((path.to_path_buf(), vec![e]))).unwrap();
                 return;
             }
         };
 
-        let sources = PartialLoader::parse(ext, &source_text)
-            .unwrap_or_else(|| vec![JavaScriptSource::new(&source_text, source_type)]);
+        let sources = PartialLoader::parse(ext, &original_source_text)
+            .unwrap_or_else(|| vec![JavaScriptSource::new(&original_source_text, source_type)]);
 
         if sources.is_empty() {
             return;
         }
 
+        let get_errors = |source_text: &str, messages: Vec<Message>| {
+            if original_source_text == source_text {
+                messages.into_iter().map(|message| message.error).collect()
+            } else {
+                messages
+                    .into_iter()
+                    .map(|message| {
+                        // TODO: Is 80 enough?
+                        let len = 80.min(source_text.len());
+                        let finder = Finder::new(&source_text[0..len]);
```

Comment:
It only has a small improve if we only check the slice of source text like this.

Pick at most 80 bytes: 
```
hyperfine -w 10 -i -n oxlint "./oxlint_1"  -n this_pr "./oxlint_2"
Benchmark 1: oxlint
  Time (mean ± σ):      14.3 ms ±   0.6 ms    [User: 37.5 ms, System: 15.0 ms]
  Range (min … max):    12.9 ms …  17.6 ms    184 runs

Benchmark 2: this_pr
  Time (mean ± σ):      14.3 ms ±   0.5 ms    [User: 37.6 ms, System: 14.7 ms]
  Range (min … max):    13.0 ms …  16.0 ms    181 runs

Summary
  this_pr ran
    1.00 ± 0.06 times faster than oxlint
```

Pick all:

```
hyperfine -w 10 -i -n oxlint "./oxlint_1"  -n this_pr "./oxlint_2"
Benchmark 1: oxlint
  Time (mean ± σ):      14.1 ms ±   0.5 ms    [User: 37.5 ms, System: 14.9 ms]
  Range (min … max):    12.9 ms …  15.5 ms    187 runs

Benchmark 2: this_pr
  Time (mean ± σ):      14.3 ms ±   0.6 ms    [User: 37.7 ms, System: 14.9 ms]
  Range (min … max):    13.1 ms …  17.8 ms    175 runs

Summary
  oxlint ran
    1.03 ± 0.06 times faster than this_pr
```